pep-0517.txt

```diff
@@ -166,6 +166,43 @@ Where the ``build-backend`` key exists, it takes precedence over
 Projects may still wish to include a ``setup.py`` for compatibility
 with tools that do not use this spec.
 
+This PEP also defines a ``python-paths`` key for use in ``pyproject.toml``, see
```

Comment:
Minor nit, I'd prefer this to be ``python-path`` for symmetry with ``PYTHONPATH`` and ``sys.path``. Not something I'd block over, but seems like it'd make sense to me.

Reply:
I agree with @dstufft, but now that we seem to be close enough to agreeing on the idea to start discussing names, what about calling this something more specific to the intended use case, like `backend-path` or `backend-import-path`? Maybe `backend-dir` to distinguish filesystem path from `module.submodule` style import paths?

`python-path` is not very meaningful by itself as a name; it makes sense because of the symmetry with the PYTHONPATH environment variable. But that symmetry sets up the expectation that it just adds arbitrary paths, with no restriction on where the backend is loaded from.

Reply:
Personally, I agree - now that you've suggested it, I'd prefer something like `backend-path` but I don't have a strong opinion. Let's see whether there's a consensus. I suspect @njsmith may have a view here ;-)

Reply:
I like `backend-path` too.

Reply:
`backend-dir` (with a string value) sounds the best to me, especially if the frontend may enforcing backend is loaded there.

`backend-path` could be confusing since it is not obvious whether the user should pass the directory containing the backend, or the file containing the backend’s entry point.

I don’t have strong opinions towards `python-path` and its variants.

Reply:
Hmm, we'd more or less agreed on multiple paths (for things like `['src', 'vendor']`) so switching to a single path might be controversial. Personally I don't care much either way, it's one for the backend developers to agree on IMO. But if we were to switch I think it would need a quick conversation on the Discourse thread.

Reply:
Something like `backend-dirs` would work as well. I am fine with multiple paths, but a multi-value with `backend-dir` as key could be confusing (like I can never remember whether it’s `install_requires` or `installs_require`…).